wasmtime-runtime/src/table.rs

```diff
@@ -32,6 +32,48 @@ impl Table {
         }
     }
 
+    /// Returns the number of allocated elements.
+    pub fn size(&self) -> u32 {
+        self.vec.len() as u32
```

Comment:
It would be good to have some kind of sanity check, such as a `try_from` here, so ensure that we never accidentally add too many elements to a table and then silently ignore some of them.